graphql-code-generator/src/main/java/io/ballerina/graphql/generator/service/comparator/FunctionSignatureComparator.java

```diff
@@ -0,0 +1,115 @@
+package io.ballerina.graphql.generator.service.comparator;
+
+import io.ballerina.compiler.syntax.tree.FunctionSignatureNode;
+import io.ballerina.compiler.syntax.tree.ParameterNode;
+import io.ballerina.compiler.syntax.tree.RequiredParameterNode;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.ballerina.graphql.generator.service.comparator.ComparatorUtils.getParameterName;
+import static io.ballerina.graphql.generator.service.comparator.ComparatorUtils.isParameterEquals;
+
+/**
+ * Utility class to store result comparing two function signatures.
+ */
+public class FunctionSignatureComparator {
+    private List<String> addedParameters;
+    private List<String> addedViolatedParameters;
+    private List<String> removedParameters;
+    private List<ParameterComparator> typeChangedParameters;
+    private List<ParameterComparator> defaultValueRemovedParameters;
+    private List<ParameterComparator> defaultValueChangedParameters;
+    private ReturnTypeDescriptorComparator returnTypeEqualityResult;
+    private final FunctionSignatureNode prevFunctionSignature;
+    private final FunctionSignatureNode nextFunctionSignature;
+
+    public FunctionSignatureComparator(FunctionSignatureNode prevFunctionSignature,
+                                       FunctionSignatureNode nextFunctionSignature) {
+        this.prevFunctionSignature = prevFunctionSignature;
+        this.nextFunctionSignature = nextFunctionSignature;
+        addedParameters = new ArrayList<>();
+        addedViolatedParameters = new ArrayList<>();
+        removedParameters = new ArrayList<>();
+        typeChangedParameters = new ArrayList<>();
+        defaultValueRemovedParameters = new ArrayList<>();
+        defaultValueChangedParameters = new ArrayList<>();
+        separateMembers();
+    }
+
+    public void separateMembers() {
```

Comment:
This method seems too long. Check all similar methods. Usually, functions this long do smell something. We should be able to break them into small pieces. Refer to the `Clean Code` about how to write functions with a single responsibility.